### DISCUSSION TITLE
generate admin view for third party packages

### DISCUSSION THREAD

**Discussion by Temian1** [filamentphp/filament#3349] [Help] [answered]
so I wanted to generate Admin for some third party packages I installed on my project but , it's not working how can I solve this please

**Comment by danharrin** [comment, marked as answer]
1) Please dont open duplicate discussions #3340
2) Your question is so vague, you've given us literally no information about which packages they are, what you want to be able to do, how do you expect us to give you a good answer?

**Reply by Temian1** [reply]
sorry about that.

the package in question is this

https://github.com/michael-rubel/laravel-couponables

I tried generating admin view for this package but it's not working , probably because the package model isn't in the project app models

is there a workaround for this

**Reply by danharrin** [reply]
So you want a Coupon resource?

`php artisan make:filament-resource Coupon`

And then in the `CouponResource` file, change the `$model` to be the package model instead

**Reply by Temian1** [reply]
yes

> So you want a Coupon resource?
> 
> `php artisan make:filament-resource Coupon`
> 
> And then in the `CouponResource` file, change the `$model` to be the package model instead

just to be sure, that means i will be able too use , only simple generate for this

because my plan is too generate the coupon tables as well using this
```
php artisan make:filament-resource Customer --generate

**Reply by danharrin** [reply]
You cant generate because that relies on the model existing, just write the columns and fields out manually and change the `$model`

**Reply by Temian1** [reply]
thanks for clarification, but I hope one day we will able to generate all relating tables and models for a third-party package
